src/PrimitiveObsession/Components.cs

```diff
@@ -0,0 +1,21 @@
+﻿using PrimitiveObsession.Config;
+
+namespace PrimitiveObsession
+{
+    public class Foo
+    {
+        public Foo(Bar bar, ConnectionString connectionString) { }
+    }
+
+    public class Bar { }
+
+    public class Baz
+    {
+        public Baz(Bar bar, string connectionString) { }
+    }
+
+    public class Qux
+    {
+        public Qux(Bar bar, string url) { }
+    }
+}
```

Comment:
Questo è un commit che non farei se questo fosse codice produttivo, ma penso che sia molto comodo a supporto del nostro articolo: tutte le classi coinvolte sono condensate, senza fronzoli, solo con gli elementi che servono per l'argomentazione, e sono tutte insieme nello stesso file, così il lettore può confrontarle con uno sguardo, comparando le rispettive registrazioni (che già sono su un unico file, nel modulo)

Ho fatto anche un'altra cosa che non ti piacerà: ho eliminato `Program`. Del resto, nell'articolo parliamo di come registrare i componenti, non di come risolverli. Insomma, secondo me il codice in `Program` era semplicemente di troppo e poco pertinente al nostro tema.